xtask/src/diff_walk.rs

```diff
@@ -155,14 +156,25 @@ fn is_compressed(path: &Path) -> Result<bool> {
 /// requires elevated permissions.
 pub fn diff_walk(orig_path: &Path) -> Result<()> {
     // Build `mount_and_walk` in release mode.
-    run_cmd(Command::new("cargo").args([
-        "build",
-        "--release",
-        "--package",
-        "xtask",
-        "--bin",
-        "mount_and_walk",
-    ]))?;
+    let path = env::var("PATH")?;
+    run_cmd(
+        Command::new("cargo")
+            .args([
+                "build",
+                "--release",
+                "--package",
+                "xtask",
+                "--bin",
+                "mount_and_walk",
+            ])
+            // Clear the env except for the PATH var. This avoids
+            // unwanted rebuilds. The details are complicated, but
```

Comment:
nit: a link or links to what you looked at to determine that the details are complicated might be helpful in the future if this workaround ends up causing its own issues.

Reply:
Added a couple links to past issues. I don't know if either one is directly relevant to how this started failing, but they at least have the right general flavor.